gitlab/data_source_gitlab_users.go

```diff
@@ -117,14 +170,76 @@ func flattenGitlabUsers(users []*gitlab.User) []interface{} {
 			"projects_limit":     user.ProjectsLimit,
 			"state":              user.State,
 			"external":           user.External,
+			"extern_uid":         user.ExternUID,
+			"organization":       user.Organization,
+			"two_factor_enabled": user.TwoFactorEnabled,
 		}
 
 		if user.CreatedAt != nil {
-			values["created_at"] = user.CreatedAt.String()
+			values["created_at"] = user.CreatedAt
 		}
 
 		usersList = append(usersList, values)
 	}
 
 	return usersList
 }
+
+func expandGitlabUsersOptions(d []interface{}) (*gitlab.ListUsersOptions, int, error) {
+	if len(d) == 0 {
+		return nil, 0, nil
+	}
+
+	data := d[0].(map[string]interface{})
+	listUsersOptions := &gitlab.ListUsersOptions{}
+	options := ""
+
+	if orderBy := data["order_by"].(string); orderBy != "" {
+		listUsersOptions.OrderBy = &orderBy
+		options += orderBy
+	}
+	if sort := data["sort"].(string); sort != "" {
+		listUsersOptions.Sort = &sort
+		options += sort
+	}
+	if search := data["search"].(string); search != "" {
+		listUsersOptions.Search = &search
+		options += search
```

Comment:
What about hashing user input to reduce the risk of not detecting changes?

I'm thinking about clashes between searches like `metrue` with `active = false` and `me` with `active = true`.

Inserting a separator like `,` between concatenated values may also help.

Reply:
I'm not sure what you mean here. I'm actually hashing user input in order so I don't see how there could be a conflict.

Reply:
I mean that, as is, the following user inputs generate the same `metrue` hash unless I'm missing something:

```
{
 search = "metrue"
 active = false
}
```

```
{
 search = "me"
 active = true
}
```

In this example, this happens because active is not added to the hash if false.

Added a separator is one workaround, but not the only one. Hashing all values should also work.